app/controllers/api/physical_servers_controller.rb

```diff
@@ -41,37 +41,77 @@ def restart_mgmt_controller_resource(type, id, _data)
     end
 
     def refresh_resource(type, id, _data = nil)
-      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" unless id
+      refresh_physical_server("refresh_ems", type, id)
+    end
 
-      ensure_resource_exists(type, id) if single_resource?
+    def apply_config_pattern_resource(type, id, data)
+      apply_config(:apply_config_pattern, type, id, data)
+    end
 
+    private
+
+    def change_resource_state(state, type, id, data = {})
+      raise BadRequestError, "Must specify an id for changing a #{type} resource" unless id
+
+      ensure_resource_exists(type, id) if single_resource?
       api_action(type, id) do |klass|
-        physical_server = resource_search(id, type, klass)
-        api_log_info("Refreshing #{physical_server_ident(physical_server)}")
-        refresh_physical_server(physical_server)
+        begin
+          server = resource_search(id, type, klass)
+          desc = "Requested server state #{state} for #{server_ident(server)}"
+          do_action(state, desc, server, data)
+        rescue StandardError => err
+          action_result(false, err.to_s)
+        end
       end
     end
 
-    private
+    def refresh_physical_server(state, type, id)
+      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" unless id
+
+      ensure_resource_exists(type, id) if single_resource?
+      api_action(type, id) do |klass|
+        begin
+          physical_server = resource_search(id, type, klass)
+          desc = "#{physical_server_ident(physical_server)} refreshing"
+          do_action(state, desc, physical_server)
+        rescue StandardError => err
+          action_result(false, err.to_s)
+        end
+      end
+    end
```

Comment:
got confused reviewing, the refresh change seems to be a refactor, should have been a separate PR or at least commit.

Reply:
Yes. The refresh change is a refactor. I'll separate this.